uk.co.compucorp.civicrm.hrcore/hrcore.php

```diff
@@ -411,22 +411,43 @@ function _hrcore_coreMenuChanges(&$params) {
     unset($params[$searchNavId]['child'][$itemId]);
   }
 
-  // remove main Reports menu
-  $reportsNavId = CRM_Core_DAO::getFieldValue('CRM_Core_DAO_Navigation', 'Reports', 'id', 'name');
-  unset($params[$reportsNavId]);
+  _hrcore_removeItemFromMainNavigationMenuByName('Reports', $params);
+  _hrcore_removeItemFromMainNavigationMenuByName('Mailings', $params);
+  _hrcore_removeItemFromMainNavigationMenuByName('Support', $params);
 
   // Remove Admin items
-  $adminNavId = CRM_Core_DAO::getFieldValue('CRM_Core_DAO_Navigation', 'Administer', 'id', 'name');
+  $adminNavId = _hrcore_getNavigationIdByName('Administer');
+  _hrcore_removeItemFromMainNavigationMenuByName('CiviReport', $params[$adminNavId]['child']);
 
-  $civiReportNavId = CRM_Core_DAO::getFieldValue('CRM_Core_DAO_Navigation', 'CiviReport', 'id', 'name');
+  $civiCaseNavId = _hrcore_getNavigationIdByName('CiviCase');
+  _hrcore_removeItemFromMainNavigationMenuByName('Redaction Rules',$params[$adminNavId]['child'][$civiCaseNavId]['child']);
+}
 
-  $civiCaseNavId = CRM_Core_DAO::getFieldValue('CRM_Core_DAO_Navigation', 'CiviCase', 'id', 'name');
-  $redactionRulesNavId = CRM_Core_DAO::getFieldValue('CRM_Core_DAO_Navigation', 'Redaction Rules', 'id', 'name');
-  $supportNavId = CRM_Core_DAO::getFieldValue('CRM_Core_DAO_Navigation', 'Support', 'id', 'name');
+/**
+ * Removes the item with the given $name from the root of the given $menu
+ *
+ * @todo Use the NavigationMenuHelper methods instead of these helper functions here
```

Comment:
I did think of these, although your suggestion in the PR description to move these to storing in the database seems like a better long term goal. In that case we can probably remove the `@todo` unless you think we're going to act on it

Reply:
I kept the `@todo`, but not it also mentions the possibility of saving the menu changes to the database